schema/common.json

```diff
@@ -4,6 +4,103 @@
   "description": "Serverless Workflow specification - common schema",
   "type": "object",
   "definitions": {
+    "expressionMarkup": {
+      "type": "string",
+      "pattern": "^\\${.*}$"
+    },
+    "jsonpointer": {
```

Comment:
https://datatracker.ietf.org/doc/html/rfc6901 ?
for this i think we need to decide do we want to use this in spec first ? :)

Reply:
First of all, this is not ghost-adding it. You've specified that the toStateData would point to elements in state data that the results should be written to. So, you've already added pointers, which are not jq expressions.
But I agree, so I removed the jsonpointer thingy. I think the outcome of merging results into state data would be much clearer if it's done in the expression language providing the result in a variable. That way, we can resolve:
- @jorgenj's issue on whether deep merge or flat merge is done
- the error in the merge description in the spec, which does not match what jq does (neither `*` nor `+` operations would neither unique-merge nor concatenate nested arrays/lists)